File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    let result = 0;
    for (let i = 0; i < discount.length - 9; i++) {
        let j;
        for (j = 0; j < want.length; j++) {
            let count = 0;
            for (let k = i; k < i + 10; k++) {
                if (discount[k] === want[j]) count ++;
            }
            if (count < number[j]) break;
        }
        if (j === want.length) result++;
    }
    return result;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
from collections import Counter

def solution(want, number, discount):
    result = 0
    want_dict = {want[i]: number[i] for i in range(len(want))}
    
    for i in range(len(discount)-9):
        discount_dict = dict(Counter(discount[i:i+10]))
        flag = True
        
        for w in want:
            if discount_dict.get(w) == None or discount_dict[w] < want_dict[w]:
                flag = False
                break
                
        if flag: result += 1
    
    return result
    